added payout docs

voith opened the pull request.

voith committed:
added payout docs

voith committed:
fixed typos

villesundell commented on docs/source/payout.rst:
```diff
@@ -0,0 +1,51 @@
+
+The command line tool supports a simple CSV input to make payouts to shareholders.
+
+* Shares are exchanged here for ether or tokens like Gemini USD.
```
Shares itself are not exchanged/given (away). If you own security tokens, you are entitled to the payouts.

villesundell commented on docs/source/payout.rst:
```diff
@@ -0,0 +1,51 @@
+* ``payout-distribute`` takes in the input csv with investor info.
+
+* ``create-holders-payout-csv`` is an optional command that can be used to generate the CSV. This CSV contains investor with
+    their respective shares. This basically fetches data by running ``token-scan`` in the background.
```
Do we need a wrapper? Could we improve `token-scan` by including CSV support?

voith replied:
These two commands have different purpose. I don't want to combine the two into one(separation of concern).

villesundell commented on docs/source/payout.rst:
```diff
@@ -0,0 +1,51 @@
+* payouts can be made either in ether or any other payable ERC20 compliant token. use `--payment-type` flag in
+  ``payout-distribute`` to specify the type of payout. ``payment-type`` can either be `ether` or `token`. If ``payment-type``
+  is `token` then ``payment-token-address`` and ``payment-token-contract-name`` are required parameters. ``payment-token-address``
+  refers to the address of the deployed payout-token and  ``payment-token-contract-name`` refers to the name of the smart contract.
```
Name (and/or symbol) could be asked from the token itself?

villesundell commented on docs/source/payout.rst:
```diff
@@ -0,0 +1,51 @@
+
+This should output:
+
+
```
Also, we were planning that address(0) would be intentionally skipped, so the total amount of tokens (and their proportional value) would not change.

voith committed:
added payout-distribute command

voith committed:
added test for create_holders_payout_csv

voith committed:
fix email and name

voith committed:
added payout-distribute test

voith committed:
payout ether test

voith committed:
test token distribution

voith committed:
[cleanup] remove unused code

voith committed:
test re-run csv

voith committed:
fix interger division

voith committed:
cleanup

voith committed:
fix broken test

voith committed:
update docs

voith committed:
fix broken test

voith committed:
fix comand line params

voith committed:
added tutorial for testing process end to end

voith committed:
add another step to tutorial

voith committed:
cosmetic fixes

voith committed:
checkout file

voith committed:
fix merge conflicts

voith committed:
clean up

voith committed:
clean up

voith committed:
resolve merge conflicts

villesundell committed:
Fixed a typo

villesundell committed:
Edited payout documentation

Edited on-chain PayoutContract interaction documentation.

villesundell committed:
kyc-manage: Supporting now the new BasicKYC

Also needed to implement --kyc-contract-address to support a scenario
where a different instance is used for whitelisting than the instance
where it was deployed.

villesundell committed:
utils: Added enum KYCAttribute for readability

Ported Solidity enum KYCAttribute to python. The original is:

  enum KYCAttribute {KYCCleared, CanPushTokens}

koirikivi committed:
Don't ignore kyc_contract_address if BasicKYC is deployed

koirikivi committed:
whitelist_kyc_address: support custom BroadcastAccount/PreparedTransaction

koirikivi committed:
whitelist_kyc_address: add flag for broadcasting

koirikivi committed:
Fix new test failure

Tests are still broken but not because of new code

koirikivi committed:
Comment-out references to removed ABI functions

koirikivi committed:
whitelist_kyc_address: don't broadcast by default

voith committed:
Merge pull request #14 from TokenMarketNet/payout-docs-fixes

Payout docs fixes

koirikivi committed:
Return tx from whitelist_kyc_address

villesundell committed:
BasicKYC: Fixed KYC deployment function

The new BasicKYC needs two extra transactions
  - One to set "setter" address
  - One to be issued from setter, clearing address 0x0 for transfer,
    which is required for minting

villesundell committed:
Merge branch 'master' into payout-docs

villesundell committed:
Temporary commit: with this, you can reach "database locked"

Merging newest master had some problems with tests.
This WIP commit solves some of the problems, but also introduces
new ones.

This commit it just for sharing work in progress.

voith commented:
closing all stale PR's
